packages/core/src/types/arazzo.ts

```diff
@@ -0,0 +1,345 @@
+    parameters: parameters,
+    successCriteria: criteriaObjects,
+    onSuccess: onSuccessList,
+    onFailure: onFailureList,
+    outputs: {
+      type: 'object',
+      additionalProperties: {
+        type: 'string',
+      },
+    },
+    'x-inherit': inherit,
+    'x-expect': expectSchema,
+    'x-assert': { type: 'string' },
+    'x-operation': extendedOperation,
+    requestBody: requestBody,
+  },
+  required: ['stepId'],
+  oneOf: [
+    { required: ['x-operation'] },
+    { required: ['operationId'] },
+    { required: ['operationPath'] },
+    { required: ['workflowId'] },
+  ],
+} as const;
+const steps = {
+  type: 'array',
+  items: step,
+} as const;
+const workflow = {
+  type: 'object',
+  properties: {
+    workflowId: { type: 'string' },
+    summary: { type: 'string' },
+    description: { type: 'string' },
+    parameters: parameters,
+    dependsOn: { type: 'array', items: { type: 'string' } },
+    inputs: {
+      type: 'object',
+      properties: {
+        type: {
+          type: 'string',
+        },
+        properties: {
+          type: 'object',
+          additionalProperties: true,
+        },
+      },
+      required: ['type'],
+    },
+    outputs: {
+      type: 'object',
+      additionalProperties: {
+        type: 'string',
+      },
+    },
+    steps: steps,
+    successActions: {
+      type: 'array',
+      items: onSuccessObject,
+    },
+    failureActions: {
+      type: 'array',
+      items: onFailureObject,
+    },
+  },
+  additionalProperties: false,
+  required: ['workflowId', 'steps'],
+} as const;
+const workflows = {
+  type: 'array',
+  items: workflow,
+} as const;
+const workflowSchema = {
+  type: 'object',
+  properties: {
+    arazzo: { type: 'string', enum: ['1.0.0'] },
+    info: infoObject,
+    sourceDescriptions: sourceDescriptionsSchema,
+    'x-parameters': parameters,
+    workflows: workflows,
+    components: {
+      type: 'object',
+      properties: {
+        inputs: {
+          type: 'object',
+          additionalProperties: {
+            type: 'object',
+            additionalProperties: true,
+            properties: {
+              type: {
+                type: 'string',
+              },
+              properties: {
+                type: 'object',
+                additionalProperties: true,
+              },
+            },
+            required: ['type'],
+          },
+        },
+        parameters: {
+          type: 'object',
+          additionalProperties: parameter,
+        },
+        successActions: {
+          type: 'object',
+          additionalProperties: onSuccessObject,
+        },
+        failureActions: {
+          type: 'object',
+          additionalProperties: onFailureObject,
+        },
+      },
+    },
+  },
+  additionalProperties: false,
+  required: ['arazzo', 'info', 'sourceDescriptions', 'workflows'],
+} as const;
+
+export const ArazzoTypes: Record<string, NodeType> = getNodeTypesFromJSONSchema(
```

Comment:
@tatomyr / @RomanHotsiy There is a question where to store the Arazzo schema.
This is the second place I duplicate it, first was Spot-exp.
I would prefer to keep it centralised.
 Best candidates are Spot-ext (looks weird to import smth from it to cli), another one is maybe @redocly/config ?

We could import schema from @redocly/config in CLI and Spot-exp and have it as a single source of truce for Arazzo schema.
Please let me know what you think about it.

Reply:
why it can't be in this repo?

Reply:
We will have duplicated schema. So there will be Arazzo JSON schema copies in Redocly/cli and Spot.